### PR TITLE
[C5] Multi-Environment API Overview feature

### PR DESCRIPTION
**Pull request by RenukaFernando**
### Proposed changes in this pull request
- Environment Overview page
image

- Set auto login (multi-environment overview) configs in local storage
image
- Render multi-environment overview page only if auto login is enabled.
- Handle confirm dialog-box before switching to another environment or version.
image
- Read lifecycle status colour from styles.
- Fix https://github.com/wso2/product-apim/issues/2299

### CODE REVIEW

**Comment by malinthaprasan** [comment]
Nice work @RenukaFernando!

Bdw regarding the second screenshot, do we need to mention about the API version as well? I guess the main impact is switching from one environment to the other.

**Comment by RenukaFernando** [comment]
Thanks, @malinthaprasan!

Even in the same environment if API version is switching it shows the above confirmation.
If it is same environment:
image

File: features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/Utils.js [hunk 1]
```diff
@@ -183,6 +212,7 @@ class Utils {
 Utils.CONST = {
     LOCAL_STORAGE_ENVIRONMENT: 'environment_publisher',
     DCR_APP_INFO: '/login/login',
+    AUTO_LOGIN_ENABLED_INFO: 'auto_login',
```
Comment (pubudu538) [inline features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/Utils.js]: Didn't we remove auto_login variable name as discussed earlier? This should be something multi-environment overview related.

Comment (RenukaFernando) [reply features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/Utils.js]: I used this to store the config in localstorage
- Set auto login (multi-environment overview) configs in local storage
image

**Comment by Mariangela** [comment]
Docs Completed.
Doc Link - https://docs.wso2.com/display/AM300/Maintaining+Multiple+Environments
Issues:
https://github.com/wso2/product-apim/issues/2753
https://github.com/wso2/product-apim/issues/2756
https://github.com/wso2/product-apim/issues/2757

**Pull request merged**